src/sentry/workflow_engine/endpoints/organization_detector_types.py

```diff
@@ -0,0 +1,51 @@
+from drf_spectacular.utils import extend_schema
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint
+from sentry.api.paginator import OffsetPaginator
+from sentry.apidocs.constants import (
+    RESPONSE_BAD_REQUEST,
+    RESPONSE_FORBIDDEN,
+    RESPONSE_NOT_FOUND,
+    RESPONSE_UNAUTHORIZED,
+)
+from sentry.apidocs.parameters import GlobalParams
+from sentry.apidocs.utils import inline_sentry_response_serializer
+from sentry.workflow_engine.models.detector import Detector
+
+
+@region_silo_endpoint
+class OrganizationDetectorTypeIndexEndpoint(OrganizationEndpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.EXPERIMENTAL,
+    }
+    owner = ApiOwner.ISSUES
+
+    @extend_schema(
+        operation_id="Fetch Detector Types",
+        parameters=[
+            GlobalParams.ORG_ID_OR_SLUG,
+        ],
+        responses={
+            201: inline_sentry_response_serializer("ListDetectorTypes", list[str]),
+            400: RESPONSE_BAD_REQUEST,
+            401: RESPONSE_UNAUTHORIZED,
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def get(self, request, organization):
+        """
+        Returns a list of detector types for a given org
+        """
+        # Note: This is a temporary stop gap until we have a detector type registry
+        distinct_types = list(Detector.objects.values_list("type", flat=True).distinct())
+        distinct_types.sort()
+
+        return self.paginate(
+            request=request,
+            queryset=distinct_types,
+            paginator_cls=OffsetPaginator,
+        )
```

Comment:
What information do we want to return about each detector type? Is the slug enough? Do we need a serializer here that looks up other information about the detector type?

Reply:
We only need the slug- this endpoint will be used to drive this UI where we list the type options, but the title and description are FE concerns

![image](https://github.com/user-attachments/assets/e201cb36-0a62-4dc7-844a-74ecd6236b4f)

see full discussion [here](https://sentry.slack.com/archives/C07H2DHMSS0/p1742323387999309)